PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java

```diff
@@ -0,0 +1,483 @@
+    }
+
+    private void validateConfiguration() {
+        updateConfiguration();
+        try {
+            webAppConfiguration.validate();
+            lblMessage.setText("");
+            buttonOK.setEnabled(true);
+        } catch (ConfigurationException e) {
+            lblMessage.setText(String.format(WARNING_MESSAGE, e.getMessage()));
+            buttonOK.setEnabled(false);
+        }
+    }
+
+    private void onCancel() {
+        dispose();
+    }
+
+    private void createWebApp() {
+        updateConfiguration();
+        ProgressManager.getInstance().run(new Task.Modal(null, "Creating New WebApp...", true) {
+            @Override
+            public void run(ProgressIndicator progressIndicator) {
+                try {
+                    progressIndicator.setIndeterminate(true);
+                    result = AzureWebAppMvpModel.getInstance().createWebApp(webAppConfiguration.getModel());
+                    dispose();
+                } catch (Exception ex) {
+                    JOptionPane.showMessageDialog(null, "Create WebApp Failed : " + ex.getMessage(), "Create WebApp " +
+                            "Failed",
+                        JOptionPane.ERROR_MESSAGE);
+                }
+            }
+        });
+    }
+
+    @Override
+    public void fillSubscription(@NotNull List<Subscription> subscriptions) {
+        cbSubscription.removeAllItems();
+        for (Subscription subscription : subscriptions) {
+            cbSubscription.addItem(subscription);
+        }
+        pack();
+    }
+
+    @Override
+    public void fillResourceGroup(@NotNull List<ResourceGroup> resourceGroups) {
+        cbExistResGrp.removeAllItems();
+        for (ResourceGroup resourceGroup : resourceGroups) {
+            cbExistResGrp.addItem(resourceGroup);
+        }
+        pack();
+    }
+
+    @Override
+    public void fillAppServicePlan(@NotNull List<AppServicePlan> appServicePlans) {
+        cbExistAppServicePlan.removeAllItems();
+        appServicePlans.stream()
+            .filter(item -> Comparing.equal(item.operatingSystem(), webAppConfiguration.getOS()))
+            .sorted(Comparator.comparing(AppServicePlan::name))
+            .forEach((plan) -> {
+                cbExistAppServicePlan.addItem(plan);
+            });
+        selectAppServicePlan();
+        pack();
+    }
+
+    @Override
+    public void fillLocation(@NotNull List<Location> locations) {
+        cbLocation.removeAllItems();
+        locations.stream()
+            .sorted(Comparator.comparing(Location::displayName))
+            .forEach((location) -> {
+                cbLocation.addItem(location);
+                if (Comparing.equal(location.name(), DEFAULT_REGION.name())) {
+                    cbLocation.setSelectedItem(location);
+                }
+            });
+        pack();
+    }
+
+    @Override
+    public void fillPricingTier(@NotNull List<PricingTier> prices) {
+        cbPricing.removeAllItems();
```

Comment:
duplicode as fillWebContainer...

Reply:
yes, use new function `fillCombobox` to do fill operation